R/model-tools.R

```diff
@@ -460,3 +464,178 @@ enw_stan_to_r <- function(
   }
   return(mod)
 }
+
+#' Set caching location for Stan models
+#'
+#' This function allows the user to set a cache location for Stan models
+#' rather than a temporary directory. This can reduce the need for model
+#' compilation on every new model run across sessions or within a session.
+#' For R version 4.0.0 and above, it's recommended to use the persistent cache
+#' as shown in the example.
+#'
+#' @param path A valid filepath representing the desired cache location.
+#'
+#' @param type A character string specifying the cache type. It can be one of
+#' "session", "persistent", or "all". Default is "session".
+#' "session" sets the cache for the current session, "persistent" writes the
+#' cache location to the user’s `.Renviron` file,  and "all" does both.
+#'
+#' @return The string of the filepath set.
+#'
+#' @family modeltools
+#' @importFrom cli cli_abort cli_alert_success cli_alert_warning
+#' @importFrom rlang arg_match
+#' @export
+#' @examplesIf interactive()
+#' # Set to local directory
+#' my_enw_cache <- enw_set_cache(file.path(tempdir(), "test"))
+#' enw_get_cache()
+#' \dontrun{
+#' # Use the package cache in R >= 4.0
+#' if (R.version.string >= "4.0.0") {
+#'  enw_set_cache(
+#'    tools::R_user_dir(package = "epinowcast", "cache"), type = "all"
+#'  )
+#'}
+#'
+#'}
+enw_set_cache <- function(path, type = c("session", "persistent", "all")) {
+
+  type <- rlang::arg_match(type)
```

Comment:
its package policy for now to do this with uncommon functions and to just use importFrom with common ones. Is that good practice more generally? I think that is debatable. This certainly follows current style though so I think we should discussion what precisely we want to do in a issue vs here.

Reply:
Personally I quite like qualifying for clarity and find `@importFrom`s somewhat hard to maintain as they require keeping track of all the uses of an imported function to know when an entry can be removed. But there are a few edge cases where qualifying is really clunky, .e.g. with ``data.table::`[` `` or `data.table::.SD`. Because of this I have ended up sticking with `@importFrom` over qualifying in most instances.

Reply:
>  of an imported function to know when an entry can be removed.

Hmm yeah fair. I think making sure they are duplicated so that you have them in every function you use something is pretty key. I have had the reverse with not importing things due to losing track of what the actual imports of the package are (because they aren't in the namespace). 

Agree that `data.table` is a big offender here (along with some others that use less) that is one of the main reasons its entirely imported (and that it is regularly used)